### PR TITLE
Adding support for default payment allocations

### PR DESCRIPTION
**Pull request by pedrodalcol**
# Critical Changes

# Changes
- Adding support for default allocations at the Payment level.

# Issues Closed

# New Metadata
- alloDefaultNotEnabled label
# Deleted Metadata

### CODE REVIEW

**Comment by pedrodalcol** [comment]
### Testing Notes

Setup:
- Enable payment allocations in custom setting (Allocations_Settings / Payment_Allocations_Enabled)
- Enable Default Allocations (Allocations_Settings / Default_Allocations_Enabled)
- Setup a Default GAU (Allocations_Settings / Default General Accounting Uni)

Verify that:
* On Payment insert, that the full Payment amount is allocates to default GAU
* When Payment amount is changed, default alloc amount is recalculated

* Validate there is only one default allocation per Payment (GAU = default GAU). Create a duplicate default GAU for a payment and verify that it is combined with the other default allocation.
* Verify that Payment allocations cannot be created if Payment Allocations are enabled but Default Allocations are not.

File: src/classes/ALLO_Allocations_TDTM.cls [hunk 1]
```diff
@@ -434,20 +510,23 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
     * @param listOpps A list of opportunities.
```
Comment (lparrott) [inline src/classes/ALLO_Allocations_TDTM.cls]: Super minor, but let's update this param line now that SObjects are allowed. Could list the objects that this is intended for?

File: src/classes/ALLO_Allocations_TDTM.cls [hunk 2]
```diff
@@ -266,11 +282,18 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         for (integer i=0; i<triggerList.size(); i++) {  
             Allocation__c allo = triggerlist[i];
 
+            //throw error if allocating at the payment level but default allocation is not enabled
+            if (!settings.Default_Allocations_Enabled__c && allo.Payment__c != null) {
```
Comment (lparrott) [inline src/classes/ALLO_Allocations_TDTM.cls]: Looks like we need to expand this error to manually created Payment Allocations as well? Using a related list, I've been able to create Payment Allocations even with Default Allocations set to false.

**Comment by annejackson** [comment]
Looks like this review is still in progress, but once done let me know and put me as the assignee in Github and the QA verifier of the item in AA. Thanks @lparrott  and @pedrodalcol

**Comment by annejackson** [comment]
Looks like those feature tests are failing/there are still some conflicts @lparrott  and @pedrodalcol  fyi

**Comment by annejackson** [comment]
Testing has passed. Will wait to merge this until all payment allocations are ready to go.

**Comment by pedrodalcol** [comment]
@force2b per our conversation, I am closing this PR.

**Pull request closed without merging**